src/js/select.js

```diff
@@ -227,22 +228,29 @@ Menu.prototype._createMenuEl = function(wrapperEl, selectEl) {
       // add row item to menu
       rowEl = doc.createElement('div');
       rowEl.textContent = loopEl.textContent;
-      rowEl._muiIndex = indexNum;
-
-      // handle selected options
-      if (loopEl.selected) {
-        rowEl.className = selectedClass;
-        selectedRow = menuEl.children.length;
-      }
 
       // handle optgroup options
       if (inGroup) jqLite.addClass(rowEl, 'mui-optgroup__option');
 
-      menuEl.appendChild(rowEl);
+      if (loopEl.disabled) {
+        // do not attach muiIndex to disable <option> elements to make them
+        // unselectable.
+        rowEl.className = disabledClass;
+      } else {
+        rowEl._muiIndex = indexNum;
+
+        // handle selected options
+        if (loopEl.selected) {
+          rowEl.className = selectedClass;
+          selectedRow = menuEl.children.length;
+        }
+      }
 
       // add to index map
       indexMap[indexNum] = rowEl;
```

Comment:
My idea is to turn `indexMap` into an Array (e.g. `itemArray`) so it's easier to iterate through the active rows.

In that scenario the line above would become:
```javascript
itemArray.push(rowEl);
```

The increment/decrement methods would look like this:
```javascript
Menu.prototype.increment = function() {
  if (this.currentPos === this.optionsArray.length - 1) return;
  
  // un-select old row
  jqLite.removeClass(this.itemArray[this.currentPos], selectedClass);

  // select new row
  this.currentPos += 1;
  jqLite.addClass(this.itemArray[this.currentPos], selectedClass);
}
```

And the selectCurrent method would look like this:
```javascript
Menu.prototype.selectCurrent = function() {
  if (this.currentPos !== this.origPos) {
    this.selectEl.selectedIndex = this.itemArray[this.currentPos].index;

    // trigger change event
    util.dispatchEvent(this.selectEl, 'change');
  }
}
```

Let me know if that makes more sense. Apologies if I'm having a hard time explaining my idea.